File: quark/test/emit/json_coersion/py/json_coersion.py
from quark_runtime import *


def main():
    json = _JSONObject();
    (json).setObjectItem((u"string"), ((_JSONObject()).setString(u"this is a string")));
    (json).setObjectItem((u"number"), ((_JSONObject()).setNumber(3.14159)));
    encoded = (json).toString();
    _println(encoded);
    dec = _JSONObject.parse(encoded);
    string = ((dec).getObjectItem(u"string")).getString();
    number = ((dec).getObjectItem(u"number")).getNumber();
    _println(string);
    _println(number);


if __name__ == "__main__":
    main()

File: quark/test/emit/json_coersion/js/json_coersion.js
var _qrt = require("quark_runtime.js");

function main() {
    var json = new _qrt.JSONObject();
    (json).setObjectItem(("string"), ((new _qrt.JSONObject()).setString("this is a string")));
    (json).setObjectItem(("number"), ((new _qrt.JSONObject()).setNumber(3.14159)));
    var encoded = (json).toString();
    _qrt.print(encoded);
    var dec = _qrt.json_from_string(encoded);
    var string = ((dec).getObjectItem("string")).getString();
    var number = ((dec).getObjectItem("number")).getNumber();
    _qrt.print(string);
    _qrt.print(number);
}
exports.main = main;

main();
